src/FSharpPlus/Data/NonEmptyList.fs

```diff
@@ -215,20 +585,298 @@ module NonEmptyList =
     /// <returns>The minimum value.</returns>
     let minBy (projection: 'T -> 'U) list = List.minBy projection (list.Head :: list.Tail)
 
+    /// <summary>Returns a list of each pair of consecutive elements in the list.</summary>
+    /// <param name="list">The input list.</param>
+    /// <returns>The result list.</returns>
+    let pairwise (list: NonEmptyList<'T>) : NonEmptyList<'T * 'T> = 
+        Seq.pairwise list |> ofSeq
+
+    /// <summary>Splits the list into two lists, containing the elements for which the given function returns
+    /// <c>true</c> and <c>false</c> respectively.</summary>
+    /// <param name="predicate">A function to test each element of the list.</param>
+    /// <param name="list">The input list.</param>
+    /// <returns>A tuple containing the two lists.</returns>
+    let partition (predicate: 'T -> bool) (list: NonEmptyList<'T>) : NonEmptyList<'T> * NonEmptyList<'T> = 
+        list |> toList |> List.partition predicate |> fun (a, b) -> (ofList a, ofList b)
```

Comment:
Running `partition` on a non-empty list doesn't always return two non-empty lists?

Reply:
Good point, I will add some checking, raise an exception, and provide a `tryPartition` function.